boot/boot_serial/src/boot_serial.c

```diff
@@ -596,8 +596,13 @@ static off_t erase_range(const struct flash_area *fap, off_t start, off_t end)
     }
 
     size = flash_sector_get_off(&sect) + flash_sector_get_size(&sect) - start;
-    BOOT_LOG_INF("Erasing range 0x%jx:0x%jx", (intmax_t)start,
-		 (intmax_t)(start + size - 1));
+#if defined(__ZEPHYR__) && defined(CONFIG_MINIMAL_LIBC) && defined(CONFIG_CBPRINTF_NANO)
```

Comment:
No. Avoid Zephyr specific `CONFIG_` things in code.
You can define two different formatting strings on top of the unit and guard them with CONIFG.
Probably using `inttypes.h` definitions for PRI would be helpful here.
Maybe we need to change the string to avoid using intmax at all if possible, at all the pointer size should be known prior to compilation.

Reply:
@de-nordic 

> No. Avoid Zephyr specific `CONFIG_` things in code.

OK. Let me think about alternative solution.